### PR TITLE
Build: Fix the aggregated javadoc docs task

### PR DESCRIPTION
**Pull request by rozza**
JAVA-2693

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -303,12 +262,61 @@ configure(subprojects.findAll { it.name != 'util' && it.name != 'mongo-java-driv
 }
 
 task docs(type: Javadoc) {
-    source subprojects.grep({ it.name != 'util' }).collect {project -> project.sourceSets.main.allJava }
-//    options = subprojects.first().javadoc.options
```
Comment (rozza) [inline build.gradle]: Options became read only in the newer (4.0?) version of gradle - so to change it you have to use the merger method.

**Comment by jyemin** [comment]
Please describe what's broken about the current build so I know what to look for.  I thought the issue was that it wasn't including class from the driver-async module (compare 3.5 and 3.6 javadoc on mongodb.github.io: the 3.5 docs have the async MongoClient, the 3.6 docs do not), but running against this patch I still don't see them included.

**Comment by rozza** [comment]
This fixes the docs task, as the options were commented out when gradle was updated.  So now none of the options set via the javadoc customizations in the build are applied to the docs task!

The reason we want to fix this task is because it is the one that *must* be used to build the aggregated docs for the gh-pages.  Note the _javadoc_ task only builds individual project javadocs.  Currently, we are missing the async docs because I imagine the wrong javadocs directory was copied to gh-pages.  The docs task creates the aggregated javadocs into the `./build/docs/`, where as it currently looks like `./mongo-java-driver/build/docs/javadocs/` is live, thus missing any async API docs.

**Comment by jyemin** [comment]
Got it.  It was my screw up not to use the docs task when publishing.

**Pull request closed without merging**